tools/debugging/channels_with_minimum_balance.py

```diff
@@ -158,13 +117,156 @@ def channel_deposit_with_the_same_node_and_token_network(deposit_queue: Joinable
+    while deposit_queue:
+        to_delete = list()
+
+        for pos, channel_deposit in enumerate(deposit_queue):
+            channel = channel_details(
+                channel_deposit.endpoint, channel_deposit.token_address, channel_deposit.partner
+            )
+
+            # The channel doesn't exist yet, wait for the transaction to be
+            # mined and the local view of the channel created.
+            if channel is None:
+                continue
+
+            deposit = necessary_deposit(channel, channel_deposit.minimum_capacity)
+            to_delete.append(pos)
+
+            if deposit:
+                current_total_deposit = int(channel["total_deposit"])
+                new_total_deposit = current_total_deposit + deposit
+                deposit_json = {"total_deposit": new_total_deposit}
+
+                log.info(f"Depositing to channel {channel_deposit}")
+
+                url_channel = (
+                    f"{channel_deposit.endpoint}/api/{API_VERSION}/"
+                    f"channels/{channel_deposit.token_address}/{channel_deposit.partner}"
+                )
+                response = requests.patch(url_channel, json=deposit_json)
+
+                assert is_successful_reponse(response), response
+            else:
+                log.info(f"Channel exists and has enough capacity {channel_deposit}")
+
+        for pos in reversed(to_delete):
+            deposit_queue.pop(pos)
+
+
+def queue_channel_open(
+    nodeaddress_to_channelopenqueue: OpenQueue,
+    nodeaddress_to_channeldepositqueue: DepositQueue,
+    channel: Dict,
+    token_address: str,
+    node_to_address: Dict,
+    node_to_endpoint: Dict,
+) -> None:
+    node1 = channel["node1"]
+    node2 = channel["node2"]
+
+    participant1 = node_to_address[node1]
+    participant2 = node_to_address[node2]
+
+    minimum_capacity1 = channel["minimum_capacity1"]
+    minimum_capacity2 = channel["minimum_capacity2"]
+
+    is_node1_with_less_work = len(nodeaddress_to_channelopenqueue[participant1]) < len(
+        nodeaddress_to_channelopenqueue[participant2]
+    )
 
-        channel = channel_details(deposit.endpoint, deposit.token_address, deposit.partner)
-        if channel is None:
-            raise RuntimeError(f"Channel does not exist! {deposit}")
-        channel_deposit_if_necessary(channel, deposit)
+    if is_node1_with_less_work:
+        channel_new = ChannelNew(
+            token_address=token_address,
+            participant=participant1,
+            partner=participant2,
+            endpoint=node_to_endpoint[node1],
+            minimum_capacity=minimum_capacity1,
+        )
+        nodeaddress_to_channelopenqueue[participant1].append(channel_new)
+
+        log.info(f"Queueing {channel_new}")
+
+        channel_deposit = ChannelDeposit(
+            token_address=token_address,
+            partner=participant1,
+            endpoint=node_to_endpoint[node2],
+            minimum_capacity=minimum_capacity1,
```

Comment:
```suggestion
            minimum_capacity=minimum_capacity2,
```

The if and else branch are very similar and it's easy to miss such mistakes. Would it be better to have an if for the variable declarations at the top and just a single code path for the actual work?

```python
if is_node1_with_less_work:
   participantA = node_to_address[node1]
   participantB = node_to_address[node2]
   ...
else:
   participantA = node_to_address[node2]
   participantB = node_to_address[node1]
   ...
```